src/Controls/src/SourceGen/RouteSyntaxReceiver.cs

```diff
@@ -0,0 +1,260 @@
+									_ = bool.TryParse(literalExpression.Token.ValueText, out implicitViewModel);
+									break;
+								case "ViewModelType" when argument.Expression is TypeOfExpressionSyntax typeOfExpression:
+									viewModelType = typeOfExpression.Type;
+									break;
+							}
+
+							if (argument.NameColon is null && argument.NameEquals is null)
+							{
+								if (routeArgument)
+								{
+									routeArgument = false;
+									route = GetRoute(cds, argument.Expression);
+								}
+								else
+								{
+									lifetime = GetLifetime(argument.Expression);
+								}
+							}
+						}
+					}
+
+					if (routes.Count == 0)
+					{
+						routes.Add(route);
+					}
+
+					_routedPages.Add(new RoutedPage(cds, routes, lifetime, implicitViewModel, viewModelType));
+				}
+
+				if (serviceAttribute is not null)
+				{
+					arguments = serviceAttribute.ArgumentList?.Arguments;
+
+					lifetime = Singleton;
+					TypeSyntax? registerFor = null;
+					var useTryAdd = false;
+
+					if (arguments is not null)
+					{
+						foreach (var argument in arguments)
+						{
+							if (argument.NameColon?.Name.Identifier.Text == "lifetime")
+							{
+								lifetime = GetLifetime(argument.Expression);
+							}
+
+							switch (argument.NameEquals?.Name.Identifier.Text)
+							{
+								case "RegisterFor" when argument.Expression is TypeOfExpressionSyntax typeOfExpression:
+									registerFor = typeOfExpression.Type;
+									break;
+								case "UseTryAdd" when argument.Expression is LiteralExpressionSyntax literalExpression:
+									_ = bool.TryParse(literalExpression.Token.ValueText, out useTryAdd);
+									break;
+							}
+
+							if (argument.Expression is not null)
+							{
+								lifetime = GetLifetime(argument.Expression);
+							}
+						}
+					}
+
+					_services.Add(new Service(cds, lifetime, registerFor, useTryAdd));
+				}
+			}
+		}
+
+		private static string GetRoute(ClassDeclarationSyntax classDeclaration, ExpressionSyntax? expression) => expression switch
+		{
+			LiteralExpressionSyntax literalExpression => $"\"{literalExpression.Token.ValueText}\"",
+			InvocationExpressionSyntax invocationExpression => ((IdentifierNameSyntax)invocationExpression.Expression).Identifier.Text switch
+			{
+				"nameof" => $"nameof(global::{GetNamespace(classDeclaration)}.{invocationExpression.ArgumentList.Arguments})",
+				_ => string.Empty,// Default route
+			},
+			MemberAccessExpressionSyntax memberAccessExpression => $"{((IdentifierNameSyntax)memberAccessExpression.Expression).Identifier.Text}.{memberAccessExpression.Name.Identifier.Text}",
+			_ => string.Empty,// Default route
+		};
+
+		private static string GetLifetime(ExpressionSyntax? expression) => expression switch
+		{
+			IdentifierNameSyntax identifierName => identifierName.Identifier.ValueText,
+			MemberAccessExpressionSyntax memberAccessExpression => memberAccessExpression.Name.Identifier.Text,
+			_ => Singleton
```

Comment:
Since the service lifetime will have the default value of Singleton returned from the default/discard clause of the switch expression, didn't consider marking it as nullable.